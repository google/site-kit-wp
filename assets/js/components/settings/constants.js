/**
 * Settings constants.
 *
 * Site Kit by Google, Copyright 2024 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { MODULE_SLUG_ADS } from '../../modules/ads/datastore/constants';
import { MODULE_SLUG_SIGN_IN_WITH_GOOGLE } from '../../modules/sign-in-with-google/datastore/constants';
import { MODULE_SLUG_READER_REVENUE_MANAGER } from '../../modules/reader-revenue-manager/datastore/constants';

export const NEW_MODULES = [
	MODULE_SLUG_ADS,
	MODULE_SLUG_READER_REVENUE_MANAGER,
	MODULE_SLUG_SIGN_IN_WITH_GOOGLE,
];

export const BETA_MODULES = [ MODULE_SLUG_SIGN_IN_WITH_GOOGLE ];

export const EXPERIMENTAL_MODULES = [];
