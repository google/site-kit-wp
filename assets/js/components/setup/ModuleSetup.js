/**
 * ModuleSetup component.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import PropTypes from 'prop-types';
import { useMount } from 'react-use';
import { useCallbackOne } from 'use-memo-one';

/**
 * WordPress dependencies
 */
import { Fragment, useCallback } from '@wordpress/element';
import { __ } from '@wordpress/i18n';

/**
 * Internal dependencies
 */
import { useSelect, useDispatch, useRegistry } from 'googlesitekit-data';
import { CORE_SITE } from '../../googlesitekit/datastore/site/constants';
import { CORE_MODULES } from '../../googlesitekit/modules/datastore/constants';
import { CORE_LOCATION } from '../../googlesitekit/datastore/location/constants';
import { deleteItem } from '../../googlesitekit/api/cache';
import { trackEvent } from '../../util';
import HelpMenu from '../help/HelpMenu';
import { Cell, Grid, Row } from '../../material-components';
import Header from '../Header';
import ModuleSetupFooter from './ModuleSetupFooter';

export default function ModuleSetup( { moduleSlug } ) {
	const { navigateTo } = useDispatch( CORE_LOCATION );

	const module = useSelect( ( select ) =>
		select( CORE_MODULES ).getModule( moduleSlug )
	);

	const registry = useRegistry();

	/**
	 * When module setup done, we redirect the user to Site Kit dashboard.
	 *
	 * @since 1.0.0
	 * @since 1.18.0 Added optional redirectURL parameter.
	 *
	 * @param {string} [redirectURL] URL to redirect to when complete. Defaults to Site Kit dashboard.
	 */
	const finishSetup = useCallbackOne(
		async ( redirectURL ) => {
			await deleteItem( 'module_setup' );

			await trackEvent(
				'moduleSetup',
				'complete_module_setup',
				moduleSlug
			);

			if ( redirectURL ) {
				navigateTo( redirectURL );
				return;
			}

			const { select, resolveSelect } = registry;
			await resolveSelect( CORE_SITE ).getSiteInfo();
			const adminURL = select( CORE_SITE ).getAdminURL(
				'googlesitekit-dashboard',
				{
					notification: 'authentication_success',
					slug: moduleSlug,
				}
			);
			navigateTo( adminURL );
		},
		[ registry, navigateTo, moduleSlug ]
	);

	const onCompleteSetup = module?.onCompleteSetup;
	const onCompleteSetupCallback = useCallback(
		() => onCompleteSetup( registry, finishSetup ),
		[ onCompleteSetup, registry, finishSetup ]
	);

	const onCancelButtonClick = useCallback( async () => {
		await trackEvent( 'moduleSetup', 'cancel_module_setup', moduleSlug );
	}, [ moduleSlug ] );

	useMount( () => {
		trackEvent( 'moduleSetup', 'view_module_setup', moduleSlug );
	} );

	if ( ! module?.SetupComponent ) {
		return null;
	}

	const { SetupComponent } = module;

	return (
		<Fragment>
			<Header>
				<HelpMenu />
			</Header>
			<div className="googlesitekit-setup">
				<Grid>
					<Row>
						<Cell size={ 12 }>
							<section className="googlesitekit-setup__wrapper">
								<Grid>
									<Row>
										<Cell size={ 12 }>
											<p className="googlesitekit-setup__intro-title">
												{ __(
													'Connect Service',
													'google-site-kit'
												) }
											</p>
											<SetupComponent
												module={ module }
												finishSetup={ finishSetup }
											/>
										</Cell>
									</Row>
								</Grid>

								<ModuleSetupFooter
									module={ module }
									onCancel={ onCancelButtonClick }
									onComplete={
										typeof onCompleteSetup === 'function'
											? onCompleteSetupCallback
											: undefined
									}
								/>
							</section>
						</Cell>
					</Row>
				</Grid>
			</div>
		</Fragment>
	);
}

ModuleSetup.propTypes = {
	moduleSlug: PropTypes.string.isRequired,
};
