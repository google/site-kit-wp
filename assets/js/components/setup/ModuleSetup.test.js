/**
 * ModuleSetup component tests.
 *
 * Site Kit by Google, Copyright 2025 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * Internal dependencies
 */
import { render } from '../../../../tests/js/test-utils';
import {
	createTestRegistry,
	provideModuleRegistrations,
	provideModules,
	provideSiteInfo,
	provideUserAuthentication,
} from '../../../../tests/js/utils';
import { MODULES_ANALYTICS_4 } from '@/js/modules/analytics-4/datastore/constants';
import { MODULE_SLUG_ANALYTICS_4 } from '@/js/modules/analytics-4/constants';
import * as analyticsFixtures from '@/js/modules/analytics-4/datastore/__fixtures__';
import ModuleSetup from './ModuleSetup';
import { CORE_SITE } from '@/js/googlesitekit/datastore/site/constants';
import { VIEW_CONTEXT_MODULE_SETUP } from '@/js/googlesitekit/constants';
import { mockLocation } from '../../../../tests/js/mock-browser-utils';
import { CORE_USER } from '@/js/googlesitekit/datastore/user/constants';

describe( 'ModuleSetup', () => {
	mockLocation();

	let registry;

	beforeEach( () => {
		registry = createTestRegistry();
		registry.dispatch( CORE_USER ).receiveGetDismissedItems( [] );
		registry.dispatch( CORE_USER ).receiveGetDismissedPrompts( {} );

		provideSiteInfo( registry );
		provideUserAuthentication( registry );
		provideModuleRegistrations( registry );
	} );

	describe( 'Analytics 4', () => {
		beforeEach( () => {
			provideModules( registry, [
				{
					slug: MODULE_SLUG_ANALYTICS_4,
					active: false,
					connected: false,
				},
			] );

			const {
				accountSummaries,
				webDataStreamsBatch,
				defaultEnhancedMeasurementSettings,
			} = analyticsFixtures;
			const accounts = accountSummaries.accountSummaries;
			const properties = accounts[ 1 ].propertySummaries;
			const accountID = accounts[ 1 ]._id;
			const propertyID = properties[ 0 ]._id;
			const webDataStreamID = webDataStreamsBatch[ propertyID ][ 0 ]._id;

			registry.dispatch( MODULES_ANALYTICS_4 ).receiveGetSettings( {
				adsConversionID: '',
			} );
			registry
				.dispatch( MODULES_ANALYTICS_4 )
				.receiveGetExistingTag( null );
			registry
				.dispatch( MODULES_ANALYTICS_4 )
				.receiveGetAccountSummaries( accountSummaries );
			registry
				.dispatch( MODULES_ANALYTICS_4 )
				.receiveGetProperty( properties[ 0 ], {
					propertyID,
				} );
			registry
				.dispatch( MODULES_ANALYTICS_4 )
				.receiveGetWebDataStreamsBatch( webDataStreamsBatch, {
					propertyIDs: [ propertyID ],
				} );
			registry
				.dispatch( MODULES_ANALYTICS_4 )
				.receiveGetEnhancedMeasurementSettings(
					{
						...defaultEnhancedMeasurementSettings,
						streamEnabled: false,
					},
					{
						propertyID,
						webDataStreamID,
					}
				);

			registry.dispatch( MODULES_ANALYTICS_4 ).selectAccount( accountID );

			registry
				.dispatch( CORE_SITE )
				.receiveGetConversionTrackingSettings( {
					enabled: false,
				} );
		} );
		describe( 'initial setup flow', () => {
			let container, waitForRegistry, queryByText, getByText;

			beforeEach( async () => {
				global.location.href =
					'http://example.com/wp-admin/admin.php?page=googlesitekit-dashboard&slug=analytics-4&reAuth=true&showProgress=true';
				( { container, waitForRegistry, queryByText, getByText } =
					render(
						<ModuleSetup moduleSlug={ MODULE_SLUG_ANALYTICS_4 } />,
						{
							registry,
							viewContext: VIEW_CONTEXT_MODULE_SETUP,
							features: [ 'setupFlowRefresh' ],
						}
					) );
				await waitForRegistry();
			} );

			it( 'should display an exit button', () => {
				expect( getByText( 'Exit setup' ) ).toBeInTheDocument();
			} );
			it( 'should display the progress indicator', () => {
				expect(
					container.querySelector(
						'.googlesitekit-progress-indicator'
					)
				).toBeInTheDocument();
			} );
			it( 'should not display the help button', () => {
				expect(
					container.querySelector(
						'.googlesitekit-help-menu__button'
					)
				).not.toBeInTheDocument();
			} );
			it( 'should not display the "Connect Service" line', () => {
				expect(
					queryByText( 'Connect Service' )
				).not.toBeInTheDocument();
			} );
			it( 'should not display the setup footer', () => {
				expect(
					container.querySelector( '.googlesitekit-setup__footer' )
				).not.toBeInTheDocument();
			} );
			it( 'should match the snpashot', () => {
				expect( container ).toMatchSnapshot();
			} );
		} );
		describe( 'not initial setup flow', () => {
			let container, waitForRegistry, queryByText, getByText;

			beforeEach( async () => {
				global.location.href =
					'http://example.com/wp-admin/admin.php?page=googlesitekit-dashboard&slug=analytics-4&reAuth=true';
				( { container, waitForRegistry, queryByText, getByText } =
					render(
						<ModuleSetup moduleSlug={ MODULE_SLUG_ANALYTICS_4 } />,
						{
							registry,
							viewContext: VIEW_CONTEXT_MODULE_SETUP,
							features: [],
						}
					) );
				await waitForRegistry();
			} );

			it( 'should display an exit button', () => {
				expect( queryByText( 'Exit setup' ) ).not.toBeInTheDocument();
			} );
			it( 'should display the progress indicator', () => {
				expect(
					container.querySelector(
						'.googlesitekit-progress-indicator'
					)
				).not.toBeInTheDocument();
			} );
			it( 'should not display the help button', () => {
				expect(
					container.querySelector(
						'.googlesitekit-help-menu__button'
					)
				).toBeInTheDocument();
			} );
			it( 'should not display the "Connect Service" line', () => {
				expect( getByText( 'Connect Service' ) ).toBeInTheDocument();
			} );
			it( 'should not display the setup footer', () => {
				expect(
					container.querySelector( '.googlesitekit-setup__footer' )
				).toBeInTheDocument();
			} );
			it( 'should match the snpashot', () => {
				expect( container ).toMatchSnapshot();
			} );
		} );
	} );
} );
