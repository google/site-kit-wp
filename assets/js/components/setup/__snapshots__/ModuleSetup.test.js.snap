// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModuleSetup Analytics 4 initial setup flow should match the snpashot 1`] = `
<div>
  <header
    class="googlesitekit-header"
  >
    <div
      class="mdc-layout-grid"
    >
      <div
        class="mdc-layout-grid__inner"
      >
        <div
          class="googlesitekit-header__logo mdc-layout-grid__cell mdc-layout-grid__cell--align-middle mdc-layout-grid__cell--span-4-desktop mdc-layout-grid__cell--span-2-tablet mdc-layout-grid__cell--span-1-phone"
        >
          <a
            aria-label="Go to dashboard"
            class="googlesitekit-cta-link googlesitekit-header__logo-link"
            href="http://example.com/wp-admin/admin.php?page=googlesitekit-dashboard"
          >
            <span
              class="googlesitekit-cta-link__contents"
            >
              <div
                aria-hidden="true"
                class="googlesitekit-logo"
              >
                <svg />
                <svg />
                <span
                  class="screen-reader-text"
                >
                  Site Kit by Google Logo
                </span>
              </div>
            </span>
          </a>
        </div>
        <div
          class="googlesitekit-header__children mdc-layout-grid__cell mdc-layout-grid__cell--align-middle mdc-layout-grid__cell--span-8-desktop mdc-layout-grid__cell--span-6-tablet mdc-layout-grid__cell--span-3-phone"
        >
          <button
            class="mdc-button mdc-button--tertiary"
            target="_self"
          >
            <span
              class="mdc-button__label"
            >
              Exit setup
            </span>
          </button>
        </div>
      </div>
    </div>
  </header>
  <div
    class="googlesitekit-subheader"
  >
    <div
      class="googlesitekit-progress-indicator"
      style="grid-template-columns: 46px repeat(6, 1fr);"
    >
      <div
        class="googlesitekit-progress-indicator__segment"
      />
      <div
        class="googlesitekit-progress-indicator__segment"
      />
      <div
        class="googlesitekit-progress-indicator__segment"
      />
      <div
        class="googlesitekit-progress-indicator__segment"
      />
      <div
        class="googlesitekit-progress-indicator__segment"
      />
      <div
        class="googlesitekit-progress-indicator__segment"
      />
    </div>
  </div>
  <div
    class="googlesitekit-setup"
  >
    <div
      class="mdc-layout-grid"
    >
      <div
        class="mdc-layout-grid__inner"
      >
        <div
          class="mdc-layout-grid__cell mdc-layout-grid__cell--span-12"
        >
          <section
            class="googlesitekit-setup__wrapper"
          >
            <div
              class="mdc-layout-grid"
            >
              <div
                class="mdc-layout-grid__inner"
              >
                <div
                  class="mdc-layout-grid__cell mdc-layout-grid__cell--span-12"
                >
                  <div
                    class="googlesitekit-setup-module googlesitekit-setup-module--analytics"
                  >
                    <div
                      class="googlesitekit-setup-module__step"
                    >
                      <h1
                        class="googlesitekit-typography googlesitekit-setup__title googlesitekit-typography--headline googlesitekit-typography--medium"
                      >
                        Set up Analytics
                      </h1>
                    </div>
                    <div
                      class="googlesitekit-setup-module__step"
                    >
                      <form
                        class="googlesitekit-analytics-setup__form"
                      >
                        <p
                          class="googlesitekit-setup-module__select_account"
                        >
                          Please select the account information below. You can change this later in your settings.
                        </p>
                        <div
                          class="googlesitekit-setup-module__inputs"
                        >
                          <div>
                            <div
                              class="mdc-select googlesitekit-analytics__select-account mdc-select--outlined"
                            >
                              <i
                                class="mdc-select__dropdown-icon"
                              />
                              <input
                                name="enhanced-select"
                                type="hidden"
                                value="987654321"
                              />
                              <div
                                aria-disabled="false"
                                class="mdc-select__selected-text"
                                tabindex="0"
                              >
                                Example Com
                              </div>
                              <div
                                class="mdc-notched-outline mdc-notched-outline--upgraded mdc-notched-outline--notched"
                              >
                                <div
                                  class="mdc-notched-outline__leading"
                                />
                                <div
                                  class="mdc-notched-outline__notch"
                                >
                                  <label
                                    class="mdc-floating-label mdc-floating-label--float-above"
                                    for="googlesitekit-select-15"
                                    style=""
                                  >
                                    Account
                                  </label>
                                </div>
                                <div
                                  class="mdc-notched-outline__trailing"
                                />
                              </div>
                            </div>
                          </div>
                          <div>
                            <div
                              class="mdc-select googlesitekit-analytics-4__select-property googlesitekit-analytics-4__select-property--loaded mdc-select--outlined"
                            >
                              <i
                                class="mdc-select__dropdown-icon"
                              />
                              <input
                                name="enhanced-select"
                                type="hidden"
                                value="1122334455"
                              />
                              <div
                                aria-disabled="false"
                                class="mdc-select__selected-text"
                                tabindex="0"
                              >
                                Example Property (1122334455)
                              </div>
                              <div
                                class="mdc-notched-outline mdc-notched-outline--upgraded mdc-notched-outline--notched"
                              >
                                <div
                                  class="mdc-notched-outline__leading"
                                />
                                <div
                                  class="mdc-notched-outline__notch"
                                >
                                  <label
                                    class="mdc-floating-label mdc-floating-label--float-above"
                                    for="googlesitekit-select-16"
                                    style=""
                                  >
                                    Property
                                  </label>
                                </div>
                                <div
                                  class="mdc-notched-outline__trailing"
                                />
                              </div>
                            </div>
                            <div
                              class="googlesitekit-setup__step-hint"
                            >
                              <p>
                                What is an Analytics property?
                              </p>
                              <span
                                class="googlesitekit-info-tooltip"
                              >
                                <svg />
                              </span>
                            </div>
                          </div>
                          <div>
                            <div
                              class="mdc-select googlesitekit-analytics-4__select-webdatastream mdc-select--outlined"
                            >
                              <i
                                class="mdc-select__dropdown-icon"
                              />
                              <input
                                name="enhanced-select"
                                type="hidden"
                                value="4000"
                              />
                              <div
                                aria-disabled="false"
                                class="mdc-select__selected-text"
                                tabindex="0"
                              >
                                Main datastream (G-155BC2366E)
                              </div>
                              <div
                                class="mdc-notched-outline mdc-notched-outline--upgraded mdc-notched-outline--notched"
                              >
                                <div
                                  class="mdc-notched-outline__leading"
                                />
                                <div
                                  class="mdc-notched-outline__notch"
                                >
                                  <label
                                    class="mdc-floating-label mdc-floating-label--float-above"
                                    for="googlesitekit-select-17"
                                    style=""
                                  >
                                    Web Data Stream
                                  </label>
                                </div>
                                <div
                                  class="mdc-notched-outline__trailing"
                                />
                              </div>
                            </div>
                            <div
                              class="googlesitekit-setup__step-hint"
                            >
                              <p>
                                What is a web data stream?
                              </p>
                              <span
                                class="googlesitekit-info-tooltip"
                              >
                                <svg />
                              </span>
                            </div>
                          </div>
                        </div>
                        <div
                          class="googlesitekit-analytics-enable-enhanced-measurement"
                        >
                          <div
                            aria-checked="true"
                            class="mdc-switch mdc-switch--checked"
                            role="switch"
                            tabindex="0"
                          >
                            <div
                              class="mdc-switch__track"
                            >
                               
                            </div>
                            <div
                              class="mdc-switch__thumb-underlay"
                            >
                              <div
                                class="mdc-switch__thumb"
                              >
                                <input
                                  checked=""
                                  class="mdc-switch__native-control"
                                  id="googlesitekit-switch-5"
                                  readonly=""
                                  role="switch"
                                  type="checkbox"
                                />
                              </div>
                            </div>
                          </div>
                          <label
                            class=""
                            for="googlesitekit-switch-5"
                          >
                            Enable enhanced measurement
                          </label>
                          <p
                            class="googlesitekit-module-settings-group__helper-text"
                          >
                            This allows you to measure interactions with your content (e.g. file downloads, form completions, video views). 
                            <a
                              aria-label="Learn more (opens in a new tab)"
                              class="googlesitekit-cta-link"
                              href="https://support.google.com/analytics/answer/9216061?hl=en-US"
                              rel="noopener noreferrer"
                              target="_blank"
                            >
                              <span
                                class="googlesitekit-cta-link__contents"
                              >
                                Learn more
                              </span>
                              <span
                                class="googlesitekit-icon-wrapper"
                                style="margin-left: 5px;"
                              >
                                <svg />
                              </span>
                            </a>
                          </p>
                        </div>
                        <p
                          class="googlesitekit-color--surfaces-on-background-variant"
                        >
                          To track how visitors interact with your site, Site Kit will enable plugin conversion tracking. You can always disable it in settings. 
                          <a
                            aria-label="Learn more (opens in a new tab)"
                            class="googlesitekit-cta-link"
                            href="https://sitekit.withgoogle.com/support/?doc=plugin-conversion-tracking"
                            rel="noopener noreferrer"
                            target="_blank"
                          >
                            <span
                              class="googlesitekit-cta-link__contents"
                            >
                              Learn more
                            </span>
                            <span
                              class="googlesitekit-icon-wrapper"
                              style="margin-left: 5px;"
                            >
                              <svg />
                            </span>
                          </a>
                        </p>
                        <div
                          class="googlesitekit-setup-module__action"
                        >
                          <button
                            class="mdc-button googlesitekit-button-icon--spinner googlesitekit-button-icon--spinner__after mdc-button--raised"
                            target="_self"
                          >
                            <span
                              class="mdc-button__label"
                            >
                              Setup
                            </span>
                          </button>
                        </div>
                      </form>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </section>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ModuleSetup Analytics 4 not initial setup flow should match the snpashot 1`] = `
<div>
  <header
    class="googlesitekit-header"
  >
    <div
      class="mdc-layout-grid"
    >
      <div
        class="mdc-layout-grid__inner"
      >
        <div
          class="googlesitekit-header__logo mdc-layout-grid__cell mdc-layout-grid__cell--align-middle mdc-layout-grid__cell--span-4-desktop mdc-layout-grid__cell--span-2-tablet mdc-layout-grid__cell--span-1-phone"
        >
          <a
            aria-label="Go to dashboard"
            class="googlesitekit-cta-link googlesitekit-header__logo-link"
            href="http://example.com/wp-admin/admin.php?page=googlesitekit-dashboard"
          >
            <span
              class="googlesitekit-cta-link__contents"
            >
              <div
                aria-hidden="true"
                class="googlesitekit-logo"
              >
                <svg />
                <svg />
                <span
                  class="screen-reader-text"
                >
                  Site Kit by Google Logo
                </span>
              </div>
            </span>
          </a>
        </div>
        <div
          class="googlesitekit-header__children mdc-layout-grid__cell mdc-layout-grid__cell--align-middle mdc-layout-grid__cell--span-8-desktop mdc-layout-grid__cell--span-6-tablet mdc-layout-grid__cell--span-3-phone"
        >
          <div
            class="googlesitekit-dropdown-menu googlesitekit-dropdown-menu__icon-menu mdc-menu-surface--anchor"
          >
            <button
              aria-controls="googlesitekit-help-menu"
              aria-expanded="false"
              aria-haspopup="menu"
              aria-label="Help"
              class="mdc-button googlesitekit-header__dropdown googlesitekit-border-radius-round googlesitekit-button-icon googlesitekit-help-menu__button mdc-button--dropdown"
              target="_self"
              title="Help"
            >
              <svg />
            </button>
            <div
              class="mdc-menu mdc-menu-surface googlesitekit-width-auto"
            >
              <ul
                aria-hidden="true"
                aria-orientation="vertical"
                class="mdc-list"
                id="googlesitekit-help-menu"
                role="menu"
                tabindex="-1"
              >
                <li
                  class="googlesitekit-help-menu-link mdc-list-item"
                  role="none"
                  tabindex="-1"
                >
                  <a
                    aria-label="Fix common issues (opens in a new tab)"
                    class="googlesitekit-cta-link mdc-list-item__text"
                    href="https://sitekit.withgoogle.com/support/?doc=fix-common-issues"
                    rel="noopener noreferrer"
                    role="menuitem"
                    tabindex="-1"
                    target="_blank"
                  >
                    <span
                      class="googlesitekit-cta-link__contents"
                    >
                      Fix common issues
                    </span>
                  </a>
                </li>
                <li
                  class="googlesitekit-help-menu-link mdc-list-item"
                  role="none"
                  tabindex="-1"
                >
                  <a
                    aria-label="Read help docs (opens in a new tab)"
                    class="googlesitekit-cta-link mdc-list-item__text"
                    href="https://sitekit.withgoogle.com/documentation/"
                    rel="noopener noreferrer"
                    role="menuitem"
                    tabindex="-1"
                    target="_blank"
                  >
                    <span
                      class="googlesitekit-cta-link__contents"
                    >
                      Read help docs
                    </span>
                  </a>
                </li>
                <li
                  class="googlesitekit-help-menu-link mdc-list-item"
                  role="none"
                  tabindex="-1"
                >
                  <a
                    aria-label="Get support (opens in a new tab)"
                    class="googlesitekit-cta-link mdc-list-item__text"
                    href="https://wordpress.org/support/plugin/google-site-kit/"
                    rel="noopener noreferrer"
                    role="menuitem"
                    tabindex="-1"
                    target="_blank"
                  >
                    <span
                      class="googlesitekit-cta-link__contents"
                    >
                      Get support
                    </span>
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </header>
  <div
    class="googlesitekit-subheader"
  />
  <div
    class="googlesitekit-setup"
  >
    <div
      class="mdc-layout-grid"
    >
      <div
        class="mdc-layout-grid__inner"
      >
        <div
          class="mdc-layout-grid__cell mdc-layout-grid__cell--span-12"
        >
          <section
            class="googlesitekit-setup__wrapper"
          >
            <div
              class="mdc-layout-grid"
            >
              <div
                class="mdc-layout-grid__inner"
              >
                <div
                  class="mdc-layout-grid__cell mdc-layout-grid__cell--span-12"
                >
                  <p
                    class="googlesitekit-setup__intro-title"
                  >
                    Connect Service
                  </p>
                  <div
                    class="googlesitekit-setup-module googlesitekit-setup-module--analytics"
                  >
                    <div
                      class="googlesitekit-setup-module__step"
                    >
                      <div
                        class="googlesitekit-setup-module__logo"
                      >
                        <svg />
                      </div>
                      <h3
                        class="googlesitekit-typography googlesitekit-setup-module__title googlesitekit-typography--headline googlesitekit-typography--small"
                      >
                        Analytics
                      </h3>
                    </div>
                    <div
                      class="googlesitekit-setup-module__step"
                    >
                      <form
                        class="googlesitekit-analytics-setup__form"
                      >
                        <p
                          class="googlesitekit-setup-module__select_account"
                        >
                          Please select the account information below. You can change this later in your settings.
                        </p>
                        <div
                          class="googlesitekit-setup-module__inputs"
                        >
                          <div>
                            <div
                              class="mdc-select googlesitekit-analytics__select-account mdc-select--outlined"
                            >
                              <i
                                class="mdc-select__dropdown-icon"
                              />
                              <input
                                name="enhanced-select"
                                type="hidden"
                                value="987654321"
                              />
                              <div
                                aria-disabled="false"
                                class="mdc-select__selected-text"
                                tabindex="0"
                              >
                                Example Com
                              </div>
                              <div
                                class="mdc-notched-outline mdc-notched-outline--upgraded mdc-notched-outline--notched"
                              >
                                <div
                                  class="mdc-notched-outline__leading"
                                />
                                <div
                                  class="mdc-notched-outline__notch"
                                >
                                  <label
                                    class="mdc-floating-label mdc-floating-label--float-above"
                                    for="googlesitekit-select-33"
                                    style=""
                                  >
                                    Account
                                  </label>
                                </div>
                                <div
                                  class="mdc-notched-outline__trailing"
                                />
                              </div>
                            </div>
                          </div>
                          <div>
                            <div
                              class="mdc-select googlesitekit-analytics-4__select-property googlesitekit-analytics-4__select-property--loaded mdc-select--outlined"
                            >
                              <i
                                class="mdc-select__dropdown-icon"
                              />
                              <input
                                name="enhanced-select"
                                type="hidden"
                                value="1122334455"
                              />
                              <div
                                aria-disabled="false"
                                class="mdc-select__selected-text"
                                tabindex="0"
                              >
                                Example Property (1122334455)
                              </div>
                              <div
                                class="mdc-notched-outline mdc-notched-outline--upgraded mdc-notched-outline--notched"
                              >
                                <div
                                  class="mdc-notched-outline__leading"
                                />
                                <div
                                  class="mdc-notched-outline__notch"
                                >
                                  <label
                                    class="mdc-floating-label mdc-floating-label--float-above"
                                    for="googlesitekit-select-34"
                                    style=""
                                  >
                                    Property
                                  </label>
                                </div>
                                <div
                                  class="mdc-notched-outline__trailing"
                                />
                              </div>
                            </div>
                          </div>
                          <div>
                            <div
                              class="mdc-select googlesitekit-analytics-4__select-webdatastream mdc-select--outlined"
                            >
                              <i
                                class="mdc-select__dropdown-icon"
                              />
                              <input
                                name="enhanced-select"
                                type="hidden"
                                value="4000"
                              />
                              <div
                                aria-disabled="false"
                                class="mdc-select__selected-text"
                                tabindex="0"
                              >
                                Main datastream (G-155BC2366E)
                              </div>
                              <div
                                class="mdc-notched-outline mdc-notched-outline--upgraded mdc-notched-outline--notched"
                              >
                                <div
                                  class="mdc-notched-outline__leading"
                                />
                                <div
                                  class="mdc-notched-outline__notch"
                                >
                                  <label
                                    class="mdc-floating-label mdc-floating-label--float-above"
                                    for="googlesitekit-select-35"
                                    style=""
                                  >
                                    Web Data Stream
                                  </label>
                                </div>
                                <div
                                  class="mdc-notched-outline__trailing"
                                />
                              </div>
                            </div>
                          </div>
                        </div>
                        <div
                          class="googlesitekit-analytics-enable-enhanced-measurement"
                        >
                          <div
                            aria-checked="true"
                            class="mdc-switch mdc-switch--checked"
                            role="switch"
                            tabindex="0"
                          >
                            <div
                              class="mdc-switch__track"
                            >
                               
                            </div>
                            <div
                              class="mdc-switch__thumb-underlay"
                            >
                              <div
                                class="mdc-switch__thumb"
                              >
                                <input
                                  checked=""
                                  class="mdc-switch__native-control"
                                  id="googlesitekit-switch-11"
                                  readonly=""
                                  role="switch"
                                  type="checkbox"
                                />
                              </div>
                            </div>
                          </div>
                          <label
                            class=""
                            for="googlesitekit-switch-11"
                          >
                            Enable enhanced measurement
                          </label>
                          <p
                            class="googlesitekit-module-settings-group__helper-text"
                          >
                            This allows you to measure interactions with your content (e.g. file downloads, form completions, video views). 
                            <a
                              aria-label="Learn more (opens in a new tab)"
                              class="googlesitekit-cta-link"
                              href="https://support.google.com/analytics/answer/9216061?hl=en-US"
                              rel="noopener noreferrer"
                              target="_blank"
                            >
                              <span
                                class="googlesitekit-cta-link__contents"
                              >
                                Learn more
                              </span>
                              <span
                                class="googlesitekit-icon-wrapper"
                                style="margin-left: 5px;"
                              >
                                <svg />
                              </span>
                            </a>
                          </p>
                        </div>
                        <p
                          class="googlesitekit-color--surfaces-on-background-variant"
                        >
                          To track how visitors interact with your site, Site Kit will enable plugin conversion tracking. You can always disable it in settings. 
                          <a
                            aria-label="Learn more (opens in a new tab)"
                            class="googlesitekit-cta-link"
                            href="https://sitekit.withgoogle.com/support/?doc=plugin-conversion-tracking"
                            rel="noopener noreferrer"
                            target="_blank"
                          >
                            <span
                              class="googlesitekit-cta-link__contents"
                            >
                              Learn more
                            </span>
                            <span
                              class="googlesitekit-icon-wrapper"
                              style="margin-left: 5px;"
                            >
                              <svg />
                            </span>
                          </a>
                        </p>
                        <div
                          class="googlesitekit-setup-module__action"
                        >
                          <button
                            class="mdc-button googlesitekit-button-icon--spinner googlesitekit-button-icon--spinner__after mdc-button--raised"
                            target="_self"
                          >
                            <span
                              class="mdc-button__label"
                            >
                              Setup
                            </span>
                          </button>
                        </div>
                      </form>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="googlesitekit-setup__footer googlesitekit-setup__footer--analytics-4"
            >
              <div
                class="mdc-layout-grid"
              >
                <div
                  class="mdc-layout-grid__inner"
                >
                  <div
                    class="mdc-layout-grid__cell mdc-layout-grid__cell--align-middle mdc-layout-grid__cell--span-6-desktop mdc-layout-grid__cell--span-4-tablet mdc-layout-grid__cell--span-2-phone"
                  >
                    <a
                      class="googlesitekit-cta-link"
                      href="http://example.com/wp-admin/admin.php?page=googlesitekit-settings"
                      id="setup-analytics-4-cancel"
                    >
                      <span
                        class="googlesitekit-cta-link__contents"
                      >
                        Cancel
                      </span>
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </section>
        </div>
      </div>
    </div>
  </div>
</div>
`;
